### PR TITLE
Calendar view

### PR DESCRIPTION
**Pull request by oliveiraigorm**

### CODE REVIEW

File: js/adminly_core-script.js [hunk 1]
```diff
@@ -1,6 +1,70 @@
+const collide = (el1, el2) => {
```
Comment (frozenbanana) [inline js/adminly_core-script.js:1]: if a function is returning boolean this here I would like to have the naming convension 

`isColliding(el1, el2)`

**Review by frozenbanana — CHANGES_REQUESTED** [review]
Have a look at the comments below.

File: js/adminly_core-script.js [hunk 2]
```diff
@@ -1,6 +1,70 @@
...
+				hexToRgb(appointmentColor)
+			) {
+				if (
+					collide(element, element.nextSibling) ||
+					collide(element, element.previousSibling)
+				)
+					element.remove();
+			}
+		});
+	});
```
[the remark this answers is not in the document]
Comment (frozenbanana) [reply js/adminly_core-script.js:49]: The hard coded DOM queryStr should be written to constants

[the remark this answers is not in the document]
Comment (frozenbanana) [reply js/adminly_core-script.js:49]: `const QuerySelectorEvents = "fc-timegrid-event-harness fc-timegrid-event-harness-inset"` Perhaps

File: js/adminly_core-script.js (hunk 1 above)
Comment (oliveiraigorm) [reply js/adminly_core-script.js]: done

File: js/adminly_core-script.js [hunk 3]
```diff
@@ -1,6 +1,82 @@
...
+		: null;
+};
+const expandEventElement = (event) => {
+	if (event.style.inset.includes("50%"))
+		event.style.inset = event.style.inset.replace("50%", "0%");
+};
+const getEventColor = (event) => {
+	return event.querySelector("a").style.backgroundColor ?? null;
+};
+const overlapEvents = () => {
```
Comment (frozenbanana) [inline js/adminly_core-script.js:34]: the function is removingOverlappingEvents right?

If so, call it `removingOverlappingEvents`

File: js/adminly_core-script.js [hunk 4]
```diff
@@ -1,6 +1,82 @@
...
+	const colEvents = document.getElementsByClassName("fc-timegrid-col-events");
+	const appointmentColor = document
+		.querySelector('[title="Appointment Slots"]')
+		.parentElement.querySelector("svg")
+		.getAttribute("fill");
+	const hexAppointmentColor = hexToRgb(appointmentColor);
+	const QuerySelectorEvents =
+		"fc-timegrid-event-harness fc-timegrid-event-harness-inset";
```
Comment (frozenbanana) [inline js/adminly_core-script.js:42]: I would like to minimize the amount of hardcoded selectors or to keep them defined together.

one option is to have small getter function for each like

```
const getEventColumns = () => document.querySelector('.fc-timegrid-col-events');
const getEventsByColumn = (column) => column.querySelector('.fc-timegrid-event-harness .fc-timegrid-event-harness-inset');
const getAppointmentColor ...
```

File: js/adminly_core-script.js [hunk 5]
```diff
@@ -1,6 +1,82 @@
...
+
+	const isOverlappingMainEvent = (event) => {
+		return (
+			getEventColor(event) === hexAppointmentColor &&
+			((isColliding(event, event.nextSibling) &&
+				getEventColor(event.nextSibling) !== hexAppointmentColor) ||
+				(isColliding(event, event.previousSibling) &&
+					getEventColor(event.previousSibling) !==
+						hexAppointmentColor))
```
Comment (frozenbanana) [inline js/adminly_core-script.js:51]: increase readability further

```
const isAppointment = (event) => getEventColor(event) === getAppointmentColor();
```

File: js/adminly_core-script.js [hunk 6]
```diff
@@ -1,6 +1,82 @@
...
+		);
+	};
+
+	colEvents.forEach((col) => {
+		const events = col.getElementsByClassName(QuerySelectorEvents);
+		events.forEach((event) => {
+			if (isOverlappingMainEvent(event)) event.remove();
+			else expandEventElement(event);
```
Comment (frozenbanana) [inline js/adminly_core-script.js:59]: Now I am nit-picking but if it there is an if-else statement that can express in one liners like this I rather use ternary operation.

`isOverlappingMainEvent(event)) ? event.remove() : expandEventElement(event)`

**Review by frozenbanana — CHANGES_REQUESTED** [review]
See comments above. 

Also it looks like you are using a formatter but why is there sometimes no newlines between functions definitions?

Make sure it is consistent with 1 newline between definitions (as I think prettier has by default)

File: js/adminly_core-script.js (hunk 5 above)
Comment (frozenbanana) [reply js/adminly_core-script.js:51]: currently the check is structured like this 

`a && (b && !c) || (d && !e) `

can you abstract the logical complexity to something more like

`a || b`

File: js/adminly_core-script.js (hunk 3 above)
Comment (oliveiraigorm) [reply js/adminly_core-script.js]: done

File: js/adminly_core-script.js (hunk 5 above)
Comment (oliveiraigorm) [reply js/adminly_core-script.js]: Changed it to
```
isAppointment(event) &&
		(isOverlappingMainEvent(event, event.nextSibling) ||
			isOverlappingMainEvent(event, event.previousSibling))
```

**Comment by oliveiraigorm** [comment]
@frozenbanana @mwalbeck  Another improvement that I can it to move the functions related to the calendar view to another file since new ones will be added and the current single file will be a mess.

**Review by frozenbanana — DISMISSED** [review]
Bravo 👏!

File: js/adminly_core-script.js [hunk 7]
```diff
@@ -1,6 +1,100 @@
...
+		const timegrid = document.querySelector(".fc-timegrid");
+		if (window.addEventListener) {
+			// Normal browsers
+			timegrid.addEventListener(
+				"DOMSubtreeModified",
+				removeOverlappingEvents,
+				false
+			);
+		} else if (window.attachEvent) {
+			// IE
```
Comment (mwalbeck) [inline js/adminly_core-script.js:96]: If this code is just to support IE, we can remove it. Nextcloud itself no longer supports IE, so there is no point in us trying to support it.

File: lib/AppInfo/Application.php [hunk 8]
```diff
@@ -56,5 +56,6 @@ public function register(IRegistrationContext $context): void
     public function boot(IBootContext $context): void
     {
         Util::addScript($this->appName, 'adminly_core-script');
+		Util::addStyle($this->appName, 'main');
```
Comment (mwalbeck) [inline lib/AppInfo/Application.php:59]: Why is this empty CSS file being added?

**Review by mwalbeck — CHANGES_REQUESTED** [review]

**Review by mwalbeck — APPROVED** [review]
Very nice!

**Pull request merged**